File: 1_encapsulation/1_python/2_method.py
class Sensor:
    def __init__(self, sensor_id, location):
        self.sensor_id = sensor_id
        self.location = location

    def read_value(self):
        return "Reading sensor value..."

    def get_status(self):
        return "Sensor status OK"

# 创建对象并调用方法
temp_sensor = Sensor(sensor_id="T1", location="Living Room")
print(temp_sensor.read_value())
print(temp_sensor.get_status())

File: 1_encapsulation/2_cpp/2_method.cc
#include <iostream>
#include <string>

class Sensor
{
public:
    Sensor(const std::string &id, const std::string &loc) : sensor_id(id), location(loc) {}
    std::string read_value()
    {
        return "Reading sensor value...";
    }
    std::string get_status()
    {
        return "Sensor status OK";
    }
    std::string sensor_id;
    std::string location;
};

int main()
{
    Sensor temp_sensor("T1", "Living Room");
    std::cout << temp_sensor.read_value() << std::endl;
    std::cout << temp_sensor.get_status() << std::endl;
    return 0;
}
